community/kernel/src/main/java/org/neo4j/kernel/impl/api/state/IndexTxStateUpdater.java

```diff
@@ -34,24 +34,25 @@
 import org.neo4j.kernel.api.schema_new.index.NewIndexDescriptor;
 import org.neo4j.kernel.impl.api.KernelStatement;
 import org.neo4j.kernel.impl.api.operations.EntityReadOperations;
-import org.neo4j.kernel.impl.api.operations.SchemaReadOperations;
 import org.neo4j.kernel.impl.api.schema.NodeSchemaMatcher;
 import org.neo4j.kernel.impl.util.Validators;
 import org.neo4j.storageengine.api.NodeItem;
 import org.neo4j.storageengine.api.PropertyItem;
+import org.neo4j.storageengine.api.StoreReadLayer;
 
 import static org.neo4j.kernel.api.properties.DefinedProperty.NO_SUCH_PROPERTY;
-import static org.neo4j.kernel.api.schema_new.SchemaDescriptorPredicates.hasProperty;
 
 public class IndexTxStateUpdater
 {
-    private final SchemaReadOperations schemaReadOps;
+    private final StoreReadLayer storeReadLayer;
     private final EntityReadOperations readOps;
     private final NodeSchemaMatcher nodeIndexMatcher;
 
-    public IndexTxStateUpdater( SchemaReadOperations schemaReadOps, EntityReadOperations readOps )
+    // We can use the StoreReadLayer directly instead of the SchemaReadOps, because we know that in transactions
```

Comment:
How do we can guarantee this? shouldn't we add some assertion to double check this?